cpp/lib/src/FileUtil.cc

```diff
@@ -514,6 +515,29 @@ bool MakeEmpty(const std::string &path) {
 }
 
 
+void TouchFileOrDie(const std::string &path) {
+    if (not Exists(path))
+        WriteStringOrDie(path, "");
+    else
+        OpenForAppendingOrDie(path);
```

Comment:
This is a nice try.  Implementing "touch" optimally is tricky.  See, for example, https://stackoverflow.com/questions/40630695/linux-modify-file-modify-access-change-time for a link to a man page with useful system calls.  (This may do for now.)